pkg/objstore/swift/swift.go

```diff
@@ -0,0 +1,267 @@
+// object name including the prefix of the inspected directory.
+func (c *Container) Iter(ctx context.Context, dir string, f func(string) error) error {
+	// Ensure the object name actually ends with a dir suffix. Otherwise we'll just iterate the
+	// object itself as one prefix item.
+	if dir != "" {
+		dir = strings.TrimSuffix(dir, DirDelim) + DirDelim
+	}
+
+	options := &objects.ListOpts{Full: false, Prefix: dir, Delimiter: DirDelim}
+	err := objects.List(c.client, c.name, options).EachPage(func(page pagination.Page) (bool, error) {
+		objectNames, err := objects.ExtractNames(page)
+		if err != nil {
+			return false, err
+		}
+		for _, objectName := range objectNames {
+			if err := f(objectName); err != nil {
+				return false, err
+			}
+		}
+
+		return true, nil
+	})
+
+	return err
+}
+
+// Get returns a reader for the given object name.
+func (c *Container) Get(ctx context.Context, name string) (io.ReadCloser, error) {
+	if name == "" {
+		return nil, fmt.Errorf("error, empty container name passed")
+	}
+	response := objects.Download(c.client, c.name, name, nil)
+	return response.Body, response.Err
+}
+
+// GetRange returns a new range reader for the given object name and range.
+func (c *Container) GetRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
+	options := objects.DownloadOpts{
+		Newest: true,
+		Range:  fmt.Sprintf("bytes=%d-%d", off, off+length-1),
+	}
+	response := objects.Download(c.client, c.name, name, options)
+	return response.Body, response.Err
+}
+
+// Exists checks if the given object exists.
+func (c *Container) Exists(ctx context.Context, name string) (bool, error) {
+	err := objects.Get(c.client, c.name, name, nil).Err
+	if err == nil {
+		return true, nil
+	}
+
+	if _, ok := err.(gophercloud.ErrDefault404); ok {
+		return false, nil
+	}
+
+	return false, err
+}
+
+// IsObjNotFoundErr returns true if error means that object is not found. Relevant to Get operations.
+func (c *Container) IsObjNotFoundErr(err error) bool {
+	_, ok := err.(gophercloud.ErrDefault404)
+	return ok
+}
+
+// Upload writes the contents of the reader as an object into the container.
+func (c *Container) Upload(ctx context.Context, name string, r io.Reader) error {
+	options := &objects.CreateOpts{Content: r}
+	res := objects.Create(c.client, c.name, name, options)
+	return res.Err
+}
+
+// Delete removes the object with the given name.
+func (c *Container) Delete(ctx context.Context, name string) error {
+	return objects.Delete(c.client, c.name, name, nil).Err
+}
+
+func (*Container) Close() error {
+	// nothing to close
+	return nil
+}
+
+func (c *Container) createContainer(name string) error {
+	return containers.Create(c.client, name, nil).Err
+}
+
+func (c *Container) deleteContainer(name string) error {
+	return containers.Delete(c.client, name).Err
+}
+
+func configFromEnv() swiftConfig {
+	c := swiftConfig{
+		AuthUrl:       os.Getenv("OS_AUTH_URL"),
```

Comment:
Not a blocker but wonder if we are sticking to SWIFT name everywhere else, might be better to do SWIFT prefix here as well? Instead OS which can mean lots of things (:

Reply:
I thought about the same... In OpenStack, the params are prefixed with `OS_` (OS=OpenStack), although the acronym is overloaded :) The OpenStack community is familiar with this notation, also CLIs & hundreds of tools typically use this notation.

We can change this, but it's one of those conventions, I think would make it easy for OpenStack folks to easily understand the params when using Thanos with Swift.

Reply:
Ah, yea i remember this now. This makes sense totally.